src/System.Net.Sockets/src/System/Net/Sockets/SocketPal.Unix.cs

```diff
@@ -743,6 +745,14 @@ public static bool TryCompleteSendTo(SafeSocketHandle socket, ReadOnlySpan<byte>
                     errorCode = SocketError.Success;
                     return true;
                 }
+
+                if (socket.SendTimeout > 0 && (Environment.TickCount64 - start) >= socket.SendTimeout)
```

Comment:
This is going to issue another syscall to get the socket send timeout, right?  That's ok from a perf perspective?

Reply:
This seems like something we should avoid unless absolutely necessary.

Reply:
I did not look at implementation but I assumed getting ticks would be cheap. This does not need to be precise and I can look for alternatives. 

We will get to this code only for partial writes. When we write completely or if Send() fails.

Reply:
Actually it's not a syscall. See https://github.com/dotnet/corefx/blob/master/src/System.Net.Sockets/src/System/Net/Sockets/SafeSocketHandle.Unix.cs#L160

Reply:
Oh, this is the safe handle, not the socket. The variable naming confused me.

Reply:
In that case, is it actually the right value? What if Socket.SendTimeout is never set... does the right thing happen here, or will this be zero?

Reply:
It defaults to -1, which means no timeout.

Reply:
Ok.